app/models/model.py

```diff
@@ -204,15 +55,83 @@ def check_num_items_holes(cls, v):
         for pt_list in v:
             for pt in pt_list:
                 if len(pt) != 3:
-                    raise ValidationError(
+                    raise ValueError(
                         'Number of floats must be 3.'
                     )
         return v
 
 
-class Ground(BaseModel):
+class ShadeEnergyPropertiesAbridged(BaseModel):
 
-    type: Enum('Ground', {'type': 'Ground'})
+    type: Enum('ShadeEnergyPropertiesAbridged', {
+               'type': 'ShadeEnergyPropertiesAbridged'})
+
+    diffuse_reflectance: float = Schema(
+        0.2,
+        ge=0,
+        le=1
+    )
+
+    specular_reflectance: float = Schema(
+        0,
+        ge=0,
+        le=1
+    )
+
+    transmittance: float = Schema(
+        0,
+        ge=0,
+        le=1
+    )
+
+    transmittance_schedule: str = Schema(
+        default=None,
+        regex=r'[A-Za-z0-9_-]',
+        min_length=1,
+        max_length=100
+    )
+
+    @validator('transmittance_schedule')
+    def check_field(cls, v, values):
+        if values['transmittance'] != "" and v != "":
+            raise ValueError(
+                'Can not specify both transmittance and transmittance_schedule.'
+            )
+
+
+class ShadePropertiesAbridged(BaseModel):
+
+    type: Enum('ShadePropertiesAbridged', {'type': 'ShadePropertiesAbridged'})
+
+    energy: ShadeEnergyPropertiesAbridged
+
+
+class Shade(BaseModel):
+
+    type: Enum('Shade', {'type': 'Shade'})
+
+    name: str = Schema(
+        ...,
+        regex=r'[A-Za-z0-9_-]',
+        min_length=1,
+        max_length=100
+    )
+
+    display_name: str = Schema(
+        default=None,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+        min_length=1,
+        max_length=100
+    )
+
+    geometry: Face3D
+
+    properties: ShadePropertiesAbridged
```

Comment:
How would you get the `ShadeProperties` that is not abridged?  Seems like this method should be renamed to `abridged_properties` or `ShadePropertiesAbridged` should be renamed to `ShadeProperties`

Reply:
I am not sure why you say that but here is the explanation for how you get abridged vs. non-abridged versions of the energy properties:

Essentially all objects across all of our Python libraries have `to-dict` and `from_dict` methods that allow you to serialize/re-serialize the objects from a dictionary.  This allows you to always take whatever you have in Python, write it to a JSON and re-seralize whenever or wherever you like.  It has uses beyond just sending objects to a server and writing objects into a format that can be parsed by the energy-model-measure. We see this complete JSON-ability of the objects as critical for being a cross-platform "interoperability hub" that can interact with the input/output of other languages.  The expectation when you simply call `to_dict` and `from_dict` is that you should be able to get a complete version of the object that you originally serialized when you re-serialize it.  However, this would obviously be impractical in a `Model` where you have many `Faces` share the same `construction`, many `Shades` share the same `transmittance_schedule`, etc.  For this reason, the `to_dict` methods of `FaceEnergyProperties` and `ShadeEnergyProperties` have an optional input argument to request an `abridged` version.  This avoids the need to a complete serialization under the assumption that things like constructions and schedules are described elsewhere in the parent object they are a part of.

So the way that you get a non-abridged version of the properties is by setting the `abridged` input of the Python `to_dict` method to `False`.  Then, you get a dictionary that can be accepted by the `from_dict` classmethod of the object.

Reply:
Thanks @chriswmackey My question here was if I have a `Shade` object (in either JSON or Python) and I get the `properties` key in JSON or call the `properties` method in Python, do I know the type of the object I will get back?  Will it always return a `ShadePropertiesAbridged` or could it be a `ShadeProperties` or a `ShadePropertiesAbridged`?  If it could return either of the two types, then I have to check which type I get back every time I get the `properties`.  Is that right?

Reply:
If `properties` always returns a `ShadePropertiesAbridged` object (which this version of the schema implies) that is good for me.  But in that case, is there a second version of the schema in which `properties` always returns a `ShadeProperties`?

Reply:
@macumber ,
Sorry for the late response to your question.  The answer is different between Python and JSON.

When you are in Python, nothing is ever abridged because object references are built into the language.  So when you call `Shade.properties.energy.construction`, it returns a whole `ShadeConstruction` object instance.  But, importantly, this object instance of the construction is the same for all of the Shade objects using it.  So there's no extra memory associated with having multiple shades using the full `ShadeConstruction` object instead of just storing the name of the `ShadeConstruction`.

This is not the case in a JSON file since, if `Shade['properties']['energy']['construction`]` returned an entire dictionary of the `ShadeConstruction`, the file size would blow up for a case where multiple `Shades` in a model use the same `ShadeConstruction`.  So whenever, the `Shade` is a part of a `Model` JSON, the `ShadeProperties` will always be abridged and the 'construction` under  `Shade['properties']['energy']['construction`]`  will always just be the name of a construction and not a full construction dictionary.  So, for all intents and purposes of the work now, `Shade['properties']` will always be `ShadePropertiesAbridged` and the only check we need to perform in the schema validation is just that the type is `ShadePropertiesAbridged` (not anything else).

When a `Shade` is not a part of a `Model` JSON, the `ShadeProperties` are usually not going to be abridged but we should not concern ourselves with this case right now since any services using this standalone shade schema are far down the line in the future. Really the only major use case for the non-abridged `ShadeProperties` right now is if someone needs to write out an individual Shade Python object to a file locally and re-serialize it later (without having to go through the process of adding the object to a Model).